src/main/java/uk/gov/hmcts/befta/DefaultTestAutomationAdapter.java

```diff
@@ -52,27 +77,33 @@ public String getNewS2SToken() {
 
     @Override
     public synchronized String getNewS2SToken(String clientId) {
-        return tokenGenerators.computeIfAbsent(clientId, key -> {
-            return getNewS2sClient(clientId);
-        }).generate();
+        String s2sToken = null;
+        try {
+            s2sToken = tokenGenerators.get(clientId, () -> getNewS2sClient(clientId)).generate();
+        } catch (ExecutionException e) {
+            BeftaUtils.defaultLog("Exception when retrieving S2S token from cache", e);
+        }
+
+        return s2sToken;
     }
 
     @Override
-    public synchronized void authenticate(UserData user, String userTokenClientId) {
-        UserData cached = users.computeIfAbsent(user.getUsername(), e -> {
-            final String accessToken = getUserAccessToken(user.getUsername(), user
-                            .getPassword(),
-                    UserTokenProviderConfig.of(userTokenClientId));
-            final AuthApi.User idamUser = idamApi.getUser(accessToken);
-            user.setId(idamUser.getId());
-            user.setAccessToken(accessToken);
-            return user;
-        });
-
-        if (user != cached) {
-            user.setId(cached.getId());
-            user.setAccessToken(cached.getAccessToken());
-        }
+    public synchronized void authenticate(UserData user, String userTokenClientId) throws ExecutionException {
+        UserData userData = users.get(user.getUsername(), () -> createUserData(user.getUsername(), user.getPassword(), userTokenClientId));
+        user.setId(userData.getId());
+        user.setAccessToken(userData.getAccessToken());
+    }
+
+    private UserData createUserData(String userName, String password, String userTokenClientId) {
```

Comment:
createUserData --> createAuthenticatedUserData

Reply:
changed to createAuthenticatedUserData